File: pdlearning/src/microservices/course-microservice/Api/Application/RequestDtos/ClassRunRequest/ChangeClassRunRescheduleStatusRequest.cs
using System;
using System.Collections.Generic;
using Microservice.Course.Domain.Enums;

namespace Microservice.Course.Application.RequestDtos
{
    public class ChangeClassRunRescheduleStatusRequest
    {
        public List<Guid> Ids { get; set; }

        public DateTime? StartDateTime { get; set; }

        public DateTime? EndDateTime { get; set; }

        public List<RescheduleSessionDto> RescheduleSessions { get; set; }

        public string Comment { get; set; }

        public ClassRunRescheduleStatus RescheduleStatus { get; set; }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/dtos/change-classrun-reschedule-status-request.ts
import { ClassRunRescheduleStatus } from '../models/classrun.model';
import { RescheduleSession } from './../models/reschedule-session.model';
export interface IClassRunRescheduleStatusRequest {
  ids: string[];
  startDateTime?: Date;
  endDateTime?: Date;
  rescheduleSessions?: RescheduleSession[];
  rescheduleStatus: ClassRunRescheduleStatus;
  comment?: string;
}
